pkg/kubectl/resource_printer.go

```diff
@@ -339,32 +340,83 @@ func podHostString(host, ip string) string {
 	return host + "/" + ip
 }
 
+// translateTimestamp returns the elapsed time since timestamp in
+// human-readable approximation.
+func translateTimestamp(timestamp util.Time) string {
+	return units.HumanDuration(time.Now().Sub(timestamp.Time))
+}
+
+// interpretContainerStatus interprets the container status and returns strings
+// associated with columns "STATUS", "CREATED", and "MESSAGE".
+// The meaning of MESSAGE varies based on the context of STATUS:
+//     STATUS: Waiting; MESSAGE: reason for waiting
+//     STATUS: Running; MESSAGE: reason for the last termination
+//     STATUS: Terminated; MESSAGE: reason for this termination
+func interpretContainerStatus(status *api.ContainerStatus) (string, string, string, error) {
+	var message string
+	if status.LastTerminationState.Termination != nil {
+		message = status.LastTerminationState.Termination.Reason
+	}
+	state := &status.State
+	if state.Waiting != nil {
```

Comment:
Yes, the intention was to expose information relevant to the last non-running status.

Alternatively, I can  expose the last termination message for all three statuses, assuming user who is curious about the waiting status would simply run `kubectl describe pod`. We get uniformity in this approach, so perhaps it's worth doing.